fatf/transparency/models/feature_influence.py

```diff
@@ -623,3 +644,406 @@ def partial_dependence(dataset: np.ndarray,
+                    scorer = check_scoring(model, 'accuracy')
+            else:
+                scorer = check_scoring(model, scoring_metric)
+        score = scorer(model, dataset, target)
+    return score
+
+
+def permutation_feature_importance(dataset: np.ndarray,
+                                   model: object,
+                                   target: np.ndarray,
+                                   as_regressor: Optional[bool] = None,
+                                   scoring_metric: Optional[str] = None,
+                                   repeat_number: Optional[int] = None):
+    '''
+    Calculates the Permutation Feature Importance (PFI)
+    of each feature in a dataset.
+
+    Based on the provided ``dataset``, ``model``
+    and ground truth ``target`` vector, this function computes the
+    Permutation Feature Importance (PFI).
+    PFI works by
+    permuting the values of each feature and measuring
+    the change in prediction error compared to the original
+    dataset. To calculate the PFI of a given dataset, first the
+    basline predictive error is calculated on the ``dataset``
+    according to a scoring metric (see below for details).
+    Next, a single feature column is permuted
+    and the predictive error is calculated for this new dataset.
+    The PFI for that feature is then the difference between the baseline
+    and permuted error. This is repeated for each feature column in the
+    dataset. The above process is repeated according to the ``repeat_number``
+    and all PFI scores are returned to user.
+
+    If scikit-learn is installed:
+        If a ``scoring_metric`` is provided by the user,
+        the function will implement the associated ``sklearn.metrics``
+        scoring function.
+        Only metrics included in ``sklearn.metrics`` are
+        compatible with this implementation.
+        If  a ``scoring_metric`` is not provided by the user,
+        to calculate the change in prediction error, the function
+        will apply the ``model``'s scikit-learn
+        scoring method if the ``model`` returns true to
+        ``fatf/transparency/sklearn/is_sklearn_model``
+        Otherwise, the model will apply a default scoring metric
+        which is either ``accuracy`` for
+        classifiers and ``max_error`` for regressors.
+        If ``as_regressor`` is specified, the
+        ``model`` will be treated accordingly.
+        If unspecified, the ``model`` will be treated as a classifer.
+
+    If scikit-learn is not installed:
+        The user is unable to input a ``scoring_metric``.
+        If ``as_regressor`` is specified
+        the ``model`` will be treated accordingly and
+        the scoring metric applied
+        will either be ``accuracy`` for classifiers or ``max_error``
+        for regressors. If the ``as_regressor`` parameter is unspecified,
+        the model will be treated as a classifier and ``accuracy``
+        will be used to generate scores.
+
```

Comment:
The current method to decide the "scoring_metric" sounds very convoluted. I wonder if we should simplify this in some way. Eg. expect a function that expects *ground_truth* and *scores*. Or check if the model has a scoring method.